test/natalie/array_test.rb

```diff
@@ -658,6 +700,96 @@
     end
   end
 
+  describe '#clear' do
+    it 'should leave an empty array empty' do
+      a = []
+      (a.clear).should === a
```

Comment:
Is this how you test the value of an expression?
And do we want to actually test / specify values of expressions?

Reply:
I don't think this would test what you want. The left side would be evaluated before the right side is passed into `should ===` Also, `===` in Ruby is kinda special for use by `case/when`, so probably not appropriate here.

I think you just want:

```
a = []
a.clear.should == []
a.should == []
```

Reply:
[Note: I deleted an earlier comment here that wasn't right.]

This is close, but `===` isn't what you want here. You should use `a.clear.should equal(a)` since that tests for object identity (same pointer).

If `===` is doing that currently, then it's a bug :wink:

Reply:
Changed it to `equal(a)` but this did work in both ruby and natalie so not sure what that means?

Reply:
Ahh, you're right. I'm so used to seeing `===` used for case/when that it looked wrong to me. Carry on :smile: